app/switchboard/Segment.scala

```diff
@@ -0,0 +1,54 @@
+package switchboard
+
+import com.typesafe.config.Config
+
+sealed abstract class Segment(val headerName: String) {
+  override def toString: String = headerName
+}
+
+object Segment {
+  // Name of the header are defined and hard-coded in Fastly VCL to be assigned
+  // to the correct proportion of traffic reflected by the name
+  case object Perc0A extends Segment("X-GU-Experiment-0perc-A")
```

Comment:
It's not quite clear to me from the naming how these segments work. What do the letters represent? Based on my initial guess, I thought I might see `Perc10-{number-between-0-and-9}`, representing 10% percent of the audience offset by the second number. But that doesn't seem to be how it works.

Reply:
TBH with you I find the logic arbitrary, probably because I'm no data scientist. For the 0, 1 and 2% tests, there can be as many as five variants (letters A to E). Someone can be in one of them, either as a control or variant, or in none at all. For 5, 10, 20 and 50% tests, there can only be one variant.